### PR TITLE
Add Style and FontAwesome

### PR DESCRIPTION
**Pull request by TomBerriot**
Changes : 
* Moves MergeRequest Components in `src/components/MergeRequest/`
* Adds Fontawesome packages
* Adds a Navbar with title and github link
* Adds style for Merge request table

Should I rename `Pill` as `Badge` maybe ?

I'm gonna fill an issue to list our need of a linter / prettier

### CODE REVIEW

File: src/components/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faGithub } from '@fortawesome/free-brands-svg-icons'
+
+class Navbar extends Component{
+  public render() {
+    return (
+        <nav className="flex items-center justify-between flex-wrap bg-blue-darkest p-6">
+            <div className="text-white mr-6">
+                <h1 className="font-semibold text-3xl tracking-tight">Gitlab Reviewer</h1>
```
Comment (nclsHart) [inline src/components/Navbar.tsx]: It should be clickable and refresh the page.

File: src/components/Pill.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react';
+
+interface Props {
+    text: string | number;
+    type: "success" | "danger" | "disable";
+}
+
+class Pill extends Component<Props> {
+  private getStyleFromType(){
```
Comment (nclsHart) [inline src/components/Pill.tsx]:
```suggestion
  private getStyleFromType() {
```

File: src/components/MergeRequest/MergeRequestItem.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  public static defaultProps = {
+    mergeRequests: []
+  };
+
+  public render() {
+    let { mergeRequest } = this.props ;
+
+    return (
+        <tr>
+          <td><a className="text-xl no-underline text-black" href={mergeRequest.web_url} target="blank">{mergeRequest.title}</a></td>
```
Comment (nclsHart) [inline src/components/MergeRequest/MergeRequestItem.tsx]:
```suggestion
          <td><a className="text-xl no-underline text-black" href={mergeRequest.web_url} target="_blank">{mergeRequest.title}</a></td>
```

Comment (nclsHart) [reply src/components/MergeRequest/MergeRequestItem.tsx]:
```suggestion
          <td><a className="text-xl no-underline text-black" href={mergeRequest.web_url} target="blank" rel="noopener noreferrer">{mergeRequest.title}</a></td>
```

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            </div>
+            <div>
+                <a href="https://github.com/TheGrowingPlant/gitlab-reviewer" target="_blank">
```
Comment (nclsHart) [inline src/components/Navbar.tsx]:
```suggestion
                <a href="https://github.com/TheGrowingPlant/gitlab-reviewer" target="_blank" rel="noopener noreferrer">
```

File: src/components/MergeRequest/MergeRequestItem.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react';
+import { MergeRequestType } from '../../types/MergeRequest';
+import Pill from '../Pill';
+
+interface Props {
+    mergeRequest: MergeRequestType;
+}
+
+interface State {
```
Comment (nclsHart) [inline src/components/MergeRequest/MergeRequestItem.tsx]: is this necessary?

File: src/components/MergeRequest/MergeRequestList.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+import MergeRequestItem from './MergeRequestItem';
+import { MergeRequestType } from '../../types/MergeRequest';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faThumbsUp, faThumbsDown } from '@fortawesome/free-solid-svg-icons'
+
+interface Props {
+    mergeRequests: MergeRequestType[];
+}
+
+interface State {
```
Comment (nclsHart) [inline src/components/MergeRequest/MergeRequestList.tsx]: Is this necessary?

File: src/components/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,22 @@
+import React, { Component } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faGithub } from '@fortawesome/free-brands-svg-icons'
+
+class Navbar extends Component{
```
Comment (nclsHart) [inline src/components/Navbar.tsx]:
```suggestion
class Navbar extends Component {
```

File: src/components/Navbar.tsx (hunk 1 above)
Comment (TomBerriot) [reply src/components/Navbar.tsx]: True :)

File: package.json [hunk 8]
```diff
@@ -32,6 +37,7 @@
     "not op_mini all"
   ],
   "devDependencies": {
+    "@fortawesome/fontawesome-free": "^5.8.1",
```
Comment (nclsHart) [inline package.json]: Why this one is in dev dependencies?

Comment (TomBerriot) [reply package.json]: I first started to read FontAwesome installation doc on "Using Package Manager" that recommended that way https://fontawesome.com/how-to-use/on-the-web/setup/using-package-managers

Then i switched on the React doc https://fontawesome.com/how-to-use/on-the-web/using-with/react

I'm gonna test if not needed and remove it if so

Comment (TomBerriot) [reply package.json]: Yes it was not needed, I removed it 👍

File: src/components/Navbar.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faGithub } from '@fortawesome/free-brands-svg-icons'
+
+class Navbar extends Component {
+  public render() {
+    return (
+        <nav className="flex items-center justify-between flex-wrap bg-blue-darkest p-6">
+            <div className="text-white mr-6">
+                <a className="text-white no-underline" href="" onClick={() => window.location.reload()}>
+                    <h1 className="font-semibold text-3xl tracking-tight">Gitlab Reviewer</h1>
```
Comment (nclsHart) [inline src/components/Navbar.tsx]:
```suggestion
                    <h1 className="font-semibold text-3xl tracking-tight">GitLab Reviewer</h1>
```

Comment (notFloran) [reply src/components/Navbar.tsx]: @TomBerriot

File: src/components/Navbar.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+                </a>
+            </div>
+            <div>
+                <a href="https://github.com/TheGrowingPlant/gitlab-reviewer" target="_blank" rel="noopener noreferrer">
+                    <FontAwesomeIcon icon={faGithub} size="2x" color="white"/>
```
Comment (notFloran) [inline src/components/Navbar.tsx]: I have mixed feelings about the link to the github repository, if we really want to put a link I would have rather seen a ribbon (see https://github.blog/2008-12-19-github-ribbons/)

**Comment by notFloran** [comment]
Good job 👍

File: src/components/Navbar.tsx (hunk 10 above)
Comment (TomBerriot) [reply src/components/Navbar.tsx]: It looks very old :/ I think writing ourselves "Fork me on [Github SVG Icon] would look better and more hand made / personal

Comment (notFloran) [reply src/components/Navbar.tsx]: The link is an example, first result on Google 😀

Comment (TomBerriot) [reply src/components/Navbar.tsx]: https://puu.sh/DgaIv/0d525a0076.png

Comment (TomBerriot) [reply src/components/Navbar.tsx]: Is that okay for you ? :p

Comment (notFloran) [reply src/components/Navbar.tsx]: I create an issue for that :  #9

**Pull request merged**